### PR TITLE
Installed Apps page revamp

### LINKED ISSUE #2571: App page install button not updating
Trying it on vercel, at least on the app page itself I get a loading state but doesn't finish on -Installed- and goes back to Install.

Also I think we should remove the label/text on the button when the loading state kicks in, so it doesn't overlap with the text.

And maybe once install is done Change to Installed (could add a checkmark icon here maybe?)

https://user-images.githubusercontent.com/6601142/164477863-7304fa1f-a3df-4bca-a1d6-7dd9fa210461.mp4

 I see a similar issue under Apps -> Installed, also [Disconnect] reappears for a few seconds after successful disconnection, Maybe we need to invalidate some query's (installed apps) just after kicking the successful disconnect so query it's updated faster?

https://user-images.githubusercontent.com/6601142/164478913-ea20cfd0-e378-48fd-b199-cd7cb27f8ba0.mp4

_Originally posted by @agustif in https://github.com/calcom/cal.com/issues/2561#issuecomment-1105274658_

### LINKED ISSUE #2715: Installed Apps is showing a lot of apps not installed
**Problem**
"Installed apps" currently shows many apps that I have not installed prompting me to "Connect". This is something I would only expect to do from within the App Store. For some apps it even says "Not Installed" in a warning, but I can't actually do anything about the issue.

**Why does this matter?**
- The majority of the apps that are showing up I have no interest in ever using (MetaMask even shows up twice!)
- It is effectively hiding installed apps that are core to my experience - primarily my calendar & conferencing apps).
Solutions

**Solution**
- Only show actually installed apps
- Put conferencing & calendars first and other categories below
- Open questions & notes
- Move embeds to settings
- Move webhooks to apps (or leave where they are?)

Communicated changes [here](https://www.figma.com/file/6sITaSOZ2JCXyKEYhC10P5/%F0%9F%96%B1-Cal-Web-App?node-id=6448%3A103354).

### PR DESCRIPTION
**Pull request by leog**
## What does this PR do?

Revamping "Installed Apps" page to just show necessary things.

Fixes #2715
Fixes #2571

Loom Video: https://www.loom.com/share/0870dda91c484e87995a0be26c7f646f
Loom Video follow-up: https://www.loom.com/share/8531837b3eb44506bcd42b9eccdc357d

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [X] Other

## How should this be tested?

- Go to "Installed Apps" page and see only installed apps shown
- New empty state would be visible if no installed app is present
- Webhooks were moved to "Developer" tab in Settings along with API Keys (moved from "Security" tab)
- Connect additional calendars section in Installed Apps page was moved to a new dropdown next to default calendar dropdown

### CODE REVIEW

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+          <p>{data.label}</p>
+        </Button>
+      )}
+    />
+  );
+};
+
+type AppOutput = inferQueryOutput<"viewer.integrations">["items"][0];
+
+const AdditionalCalendarSelector = ({ isLoading }: Props): JSX.Element | null => {
```
Comment (leog) [inline apps/web/components/AdditionalCalendarSelector.tsx:35]: New selector for additional connections

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+      query={query}
+      success={({ data }) => {
+        const options = data.items.filter(installedFilter).map((item) => ({
+          value: item.slug,
+          label: item.name,
+          image: item.imageSrc,
+          type: item.type,
+        }));
+        const customStyles = {
+          // Forgive me father for I have sinned, "any" is necessary here
```
Comment (leog) [inline apps/web/components/AdditionalCalendarSelector.tsx:51]: 😇

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import { trpc, inferQueryOutput } from "@lib/trpc";
+
+interface Props {
+  isLoading?: boolean;
+}
+
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
+  return (
+    <InstallAppButton
```
Comment (leog) [inline apps/web/components/AdditionalCalendarSelector.tsx:20]: Kudos @zomars, the Install buttons are very flexible!

File: apps/web/components/DestinationCalendarSelector.tsx [hunk 4]
```diff
@@ -50,6 +50,21 @@ const DestinationCalendarSelector = ({
         value: `${cal.integration}:${cal.externalId}`,
       })),
     })) ?? [];
+  const customStyles = {
```
Comment (leog) [inline apps/web/components/DestinationCalendarSelector.tsx:53]: Taking the opportunity to fix colors in the destination calendar selector as well

File: apps/web/components/integrations/CalendarListContainer.tsx [hunk 5]
```diff
@@ -207,39 +210,44 @@ export function CalendarListContainer(props: { heading?: false }) {
...
-                isLoading={mutation.isLoading}
-                value={query.data?.destinationCalendar?.externalId}
-              />
-            </div>
-          }
-        />
-      )}
-      <ConnectedCalendarsList onChanged={onChanged} />
-      {!!query.data?.connectedCalendars.length && (
-        <ShellSubHeading
```
Comment (leog) [inline apps/web/components/integrations/CalendarListContainer.tsx:237]: This is what was moved to the new dropdown

File: apps/web/components/webhook/WebhookListContainer.tsx [hunk 6]
```diff
@@ -23,45 +22,33 @@ export type WebhookListContainerType = {
...
+    : trpc.useQuery(["viewer.webhook.list"]);
   const [newWebhookModal, setNewWebhookModal] = useState(false);
   const [editModalOpen, setEditModalOpen] = useState(false);
   const [editing, setEditing] = useState<TWebhook | null>(null);
   return (
     <QueryCell
       query={query}
       success={({ data }) => (
-        <>
-          <ShellSubHeading className="mt-10" title={props.title} subtitle={props.subtitle} />
```
Comment (leog) [inline apps/web/components/webhook/WebhookListContainer.tsx:41]: Simplifying the UI here to comply with figma designs

File: apps/web/components/webhook/WebhookListItem.tsx [hunk 7]
```diff
@@ -41,8 +41,8 @@ export default function WebhookListItem(props: { webhook: TWebhook; onEditWebhoo
                 <span
                   key={ind}
                   className={classNames(
-                    "w-max rounded-sm px-1 text-xs ",
-                    props.webhook.active ? "bg-blue-100 text-blue-700" : "bg-blue-50 text-blue-200"
+                    "w-max rounded-sm px-1 text-xs",
+                    props.webhook.active ? "text-grey-200 bg-gray-200" : "bg-grey-50 text-grey-200"
```
Comment (leog) [inline apps/web/components/webhook/WebhookListItem.tsx:45]: Asked Ciarán if this blue was intended and he suggested to go for gray, so I changed it

File: apps/web/pages/apps/installed.tsx [hunk 8]
```diff
@@ -81,122 +81,92 @@ function ConnectOrDisconnectIntegrationButton(props: {
...
+
+interface IntegrationsContainerProps {
+  variant: App["variant"];
+  className?: string;
+}
+
+const IntegrationsContainer = ({ variant, className = "" }: IntegrationsContainerProps): JSX.Element => {
   const { t } = useLocale();
-  const query = trpc.useQuery(["viewer.integrations"], { suspense: true });
+  const query = trpc.useQuery(["viewer.integrations", { variant }], { suspense: true });
```
Comment (leog) [inline apps/web/pages/apps/installed.tsx:93]: As mentioned, we only get the integrations per variant

File: apps/web/pages/apps/installed.tsx [hunk 9]
```diff
@@ -81,122 +81,92 @@ function ConnectOrDisconnectIntegrationButton(props: {
...
+  const installedFilter = (app: AppOutput) => app.credentialIds.length > 0 || app.isGlobal;
```
Comment (leog) [inline apps/web/pages/apps/installed.tsx:94]: And we need to make sure we filter out not installed apps, as the Installed Apps page was repurposed to only show that

File: apps/web/pages/apps/installed.tsx [hunk 10]
```diff
@@ -250,21 +220,46 @@ function Web3ConnectBtn() {
...
+                {data.total > 0 ? (
+                  <>
+                    <IntegrationsContainer variant="conferencing" />
+                    <CalendarListContainer />
+                    <IntegrationsContainer variant="payment" className="mt-8" />
+                    <IntegrationsContainer variant="other" className="mt-8" />
+                    <Web3Container />
+                  </>
+                ) : (
+                  <EmptyScreen
```
Comment (leog) [inline apps/web/pages/apps/installed.tsx:246]: New empty state

File: apps/web/server/routers/viewer.tsx [hunk 11]
```diff
@@ -620,27 +624,11 @@ const loggedInViewerRouter = createProtectedRouter()
         })
       );
       // `flatMap()` these work like `.filter()` but infers the types correctly
-      const conferencing = apps.flatMap((item) => (item.variant === "conferencing" ? [item] : []));
-      const payment = apps.flatMap((item) => (item.variant === "payment" ? [item] : []));
-      const other = apps.flatMap((item) => (item.variant.startsWith("other") ? [item] : []));
-      const calendar = apps.flatMap((item) => (item.variant === "calendar" ? [item] : []));
+      const flapApps = apps.flatMap((item) => (item.variant === variant ? [item] : []));
```
Comment (leog) [inline apps/web/server/routers/viewer.tsx:627]: Changes to only return the requested variant

File: apps/web/pages/availability/index.tsx [hunk 12]
```diff
@@ -2,12 +2,12 @@ import { ClockIcon } from "@heroicons/react/outline";
 
 import { useLocale } from "@calcom/lib/hooks/useLocale";
 import showToast from "@calcom/lib/notification";
+import EmptyScreen from "@calcom/ui/EmptyScreen";
```
Comment (leog) [inline apps/web/pages/availability/index.tsx:5]: Took the opportunity to completely move EmptyScreen to monorepo

File: apps/web/pages/settings/developer.tsx [hunk 13]
```diff
@@ -0,0 +1,19 @@
...
+import WebhookListContainer from "@components/webhook/WebhookListContainer";
+
+export default function Settings() {
+  const { t } = useLocale();
+
+  return (
+    <Shell heading={t("developer")} subtitle={t("manage_developer_settings")}>
+      <SettingsShell>
+        <WebhookListContainer title={t("webhooks")} subtitle={t("receive_cal_meeting_data")} />
+        <ApiKeyListContainer />
```
Comment (leog) [inline apps/web/pages/settings/developer.tsx:15]: As mentioned in the chat, and approved by Ciarán, API Keys live in Developers tab in Settings now

File: apps/web/playwright/fixtures/users.ts [hunk 14]
```diff
@@ -145,12 +145,12 @@ export async function login(
 }
 
 export async function getPaymentCredential(page: Page) {
-  await page.goto("/apps/installed");
+  await page.goto("/apps/stripe");
```
Comment (leog) [inline apps/web/playwright/fixtures/users.ts:148]: You can't install anything from "Installed Apps" anymore, making it so it goes directly to Stripe app page

File: apps/web/playwright/integrations.test.ts [hunk 15]
```diff
@@ -24,7 +24,7 @@ test.describe("Integrations", () => {
     const user = await users.create();
     const [eventType] = user.eventTypes;
     await user.login();
-    await page.goto("/apps/installed");
+    await page.goto("/settings/developer");
```
Comment (leog) [inline apps/web/playwright/integrations.test.ts:27]: Webhook is under Developer tab in Settings now

File: apps/web/public/static/locales/en/common.json [hunk 16]
```diff
@@ -771,7 +776,6 @@
   "api_keys": "API Keys",
   "api_key_modal_subtitle": "API keys allow you to make API calls for your own account.",
   "api_keys_subtitle": "Generate API keys to use for accessing your own account.",
-  "generate_new_api_key": "Generate new API key",
```
Comment (leog) [inline apps/web/public/static/locales/en/common.json:774]: To be consistent with Webhooks section in Developer tab in Settings, I removed the text along side the "plus" icon

**Comment by leog** [comment]
Making the loom video I noticed that now that integration apps get fetched separately due to changes I introduced, the loading experience is a bit weird, maybe I can go back to what was before, thoughts?

**Review by PeerRich — APPROVED** [review]
looks good, do you have a screenshot of the empty page?

**Comment by PeerRich** [comment]
you can still add multiple google calendars right?

**Comment by leog** [comment]
> you can still add multiple google calendars right?

Of course!

**Comment by zomars** [comment]
Does this make Daily video opt-in as well? Do we need to install it from the App Store to use it as a location? Isn't the Expected behavior to be installed by default?

**Comment by leog** [comment]
> Does this make Daily video opt-in as well? Do we need to install it from the App Store to use it as a location? Isn't the Expected behavior to be installed by default?

All very good questions. This doesn't change the status for Daily video. An app is considered installed when credentials exist in the database; we could use this PR to, for once and for all, clear things up regarding the concepts of installing, connecting, adding and the `installed` flag which I think I saw some deprecation notice somewhere. Either way, additional effort will be required to achieve consistency for the Daily video app, I'm up for it! @PeerRich @Jaibles please advice.

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 2 above)
Comment (alannnc) [reply apps/web/components/AdditionalCalendarSelector.tsx:53]: I think it would be good we use this comment to explain why any it's necessary here so in the future if another one takes up this code can understand easily what its happening here

File: apps/web/pages/apps/installed.tsx [hunk 17]
```diff
@@ -96,122 +95,96 @@ function ConnectOrDisconnectIntegrationButton(props: {
...
+      }}></QueryCell>
   );
-}
+};
 
 function Web3Container() {
   const { t } = useLocale();
-
+  const result = trpc.useQuery(["viewer.web3Integration"]);
+  const isWeb3Active = result.data?.isWeb3Active as boolean;
```
Comment (alannnc) [inline apps/web/pages/apps/installed.tsx:156]:
```suggestion
  const isWeb3Active = !!result.data?.isWeb3Active;
```
If you want to convert it to strick boolean you can use double negation

**Comment by Jaibles** [comment]
We can have Daily pre installed. Overall the goal was not to start with a blank canvas here, but just to not show things that are not installed so we can pre-install Daily.

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 2 above)
Comment (leog) [reply apps/web/components/AdditionalCalendarSelector.tsx:53]: I actually tried to make it not `any` but I couldn't find a type to cover it. This is an object from `react-select` library to handle custom styles over their exported Select element. It's not that important, but I'm happy to include that description if we think it helps.

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 18]
```diff
@@ -0,0 +1,80 @@
...
+              marginLeft: "3px",
+            };
+          },
+        };
+        return (
+          <Select
+            name={"additionalCalendar"}
+            placeholder={t("connect_additional_calendar")}
+            options={options}
+            styles={customStyles}
```
Comment (zomars) [inline apps/web/components/AdditionalCalendarSelector.tsx:68]: Just inline it. We're not receiving benefit for extracting this to a variable and losing type inference.

```suggestion
            styles={{
              placeholder: (defaultStyles) => {
                return {
                  ...defaultStyles,
                  color: "#3E3E3E",
                  marginLeft: "3px",
                };
              },
            }}
```

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 19]
```diff
@@ -0,0 +1,80 @@
...
+
+import { QueryCell } from "@lib/QueryCell";
+import { trpc, inferQueryOutput } from "@lib/trpc";
+
+interface Props {
+  isLoading?: boolean;
+}
+
+const ImageOption = (props: OptionProps) => {
+  const data = props.data as { [key: string]: string; type: App["type"] };
```
Comment (zomars) [inline apps/web/components/AdditionalCalendarSelector.tsx:18]: Try using the generic instead of casting first:

```suggestion
const ImageOption = (props: OptionProps<{ [key: string]: string; type: App["type"] }>) => {
  const { data } = props;
```

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 2 above)
Comment (zomars) [reply apps/web/components/AdditionalCalendarSelector.tsx:53]: There's no benefit extracting the variable here. Addressed on my other comment.

File: apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx [hunk 20]
```diff
@@ -23,21 +24,24 @@ export default function ApiKeyListContainer() {
...
+                onClick={() => setNewApiKeyModal(true)}
+                data-testid="new_webhook"
+              />
+            }
+          />
 
           {data.length > 0 && (
-            <List className="pb-6">
+            <List className="mt-6">
               {data.map((item: any) => (
```
Comment (zomars) [inline apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx]: Pushed a fix for this any

File: apps/web/pages/apps/installed.tsx (hunk 8 above)
Comment (zomars) [reply apps/web/pages/apps/installed.tsx:111]: Can we add an input to the query it only returns installed apps? Instead of filtering on the frontend, let's just reduce the data returned from the server.

File: apps/web/pages/apps/installed.tsx [hunk 21]
```diff
@@ -96,122 +95,96 @@ function ConnectOrDisconnectIntegrationButton(props: {
...
+                        />
+                      }
+                    />
+                  ))}
+                </List>
+              </div>
+            )}
+          </>
+        );
+      }}></QueryCell>
```
Comment (zomars) [inline apps/web/pages/apps/installed.tsx:149]:
```suggestion
      }}/>
```

File: apps/web/pages/apps/installed.tsx (hunk 17 above)
Comment (zomars) [reply apps/web/pages/apps/installed.tsx:156]: What @alannnc said

File: apps/web/pages/apps/installed.tsx [hunk 22]
```diff
@@ -265,21 +238,45 @@ function Web3ConnectBtn() {
...
+                        </Button>
+                      </>
+                    }
+                  />
+                )}
+              </ClientSuspense>
+            </AppsShell>
+          </Shell>
+        );
+      }}></QueryCell>
```
Comment (zomars) [inline apps/web/pages/apps/installed.tsx:280]:
```suggestion
      }}/>
```

File: apps/web/server/routers/viewer.tsx [hunk 23]
```diff
@@ -620,27 +624,13 @@ const loggedInViewerRouter = createProtectedRouter()
         })
       );
       // `flatMap()` these work like `.filter()` but infers the types correctly
-      const conferencing = apps.flatMap((item) => (item.variant === "conferencing" ? [item] : []));
-      const payment = apps.flatMap((item) => (item.variant === "payment" ? [item] : []));
-      const other = apps.flatMap((item) => (item.variant.startsWith("other") ? [item] : []));
-      const calendar = apps.flatMap((item) => (item.variant === "calendar" ? [item] : []));
+      let flatApps = apps;
```
Comment (zomars) [inline apps/web/server/routers/viewer.tsx:627]: You can just add a `let` to `const apps` above.

**Review by zomars — CHANGES_REQUESTED** [review]
Impressive work @leog Let's just address some quality improvements first. 🙏🏽

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 1 above)
Comment (leog) [reply apps/web/components/AdditionalCalendarSelector.tsx:33]: Seems like the Apple Calendar Install Button returns the dialog to be opened, which never gets a chance to be opened as the dropdown list where the install button is included (as well as the dialog component) gets closed upon option selection. Tried a bunch of ways, got very very close, but I think the code would be too hard to follow even if we can make it work. Any suggestions on how to handle the extra element returned by the Install Button? Would it be too bad of a UX redirecting the user to some place with the dialog opened as a page? Or perhaps we can make it so the dialog becomes some kind of tooltip in this dropdown case 🤔 @Jaibles your input would be highly appreciated!

File: apps/web/components/webhook/WebhookListContainer.tsx [hunk 24]
```diff
@@ -21,47 +19,34 @@ export type WebhookListContainerType = {
 };
 
 export default function WebhookListContainer(props: WebhookListContainerType) {
-  const { t } = useLocale();
   const query = props.eventTypeId
-    ? trpc.useQuery(["viewer.webhook.list", { eventTypeId: props.eventTypeId }], {
-        suspense: true,
```
[the remark this answers is not in the document]
Comment (leog) [reply apps/web/components/webhook/WebhookListContainer.tsx:27]: Sorry, removed by accident

File: apps/web/pages/apps/installed.tsx (hunk 8 above)
Comment (leog) [reply apps/web/pages/apps/installed.tsx]: Makes sense!

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 2 above)
Comment (leog) [reply apps/web/components/AdditionalCalendarSelector.tsx]: Gotcha!

File: apps/web/components/DestinationCalendarSelector.tsx [hunk 25]
```diff
@@ -50,6 +50,22 @@ const DestinationCalendarSelector = ({
...
+    // Forgive me father for I have sinned, "any" is necessary here
```
[the remark this answers is not in the document]
Comment (leog) [reply apps/web/components/DestinationCalendarSelector.tsx]: Yep, totally

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 1 above)
Comment (zomars) [reply apps/web/components/AdditionalCalendarSelector.tsx:33]: For now it would be better to redirect to `/apps/apple-calendar` than doing nothing at all...

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 26]
```diff
@@ -0,0 +1,70 @@
...
+          image: item.imageSrc,
+          type: item.type,
+        }));
+        return (
+          <Select
+            name={"additionalCalendar"}
+            placeholder={t("connect_additional_calendar")}
+            options={options}
+            styles={{
+              placeholder: (defaultStyles: any) => ({
```
Comment (zomars) [inline apps/web/components/AdditionalCalendarSelector.tsx:53]:
```suggestion
              placeholder: (defaultStyles) => ({
```

File: apps/web/components/AdditionalCalendarSelector.tsx [hunk 27]
```diff
@@ -0,0 +1,70 @@
...
+import Select from "react-select";
+import { OptionProps } from "react-select";
+
+import { InstallAppButton } from "@calcom/app-store/components";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { App } from "@calcom/types/App";
+import { Button } from "@calcom/ui";
+
+import { QueryCell } from "@lib/QueryCell";
+import { trpc, inferQueryOutput } from "@lib/trpc";
```
Comment (zomars) [inline apps/web/components/AdditionalCalendarSelector.tsx:11]:
```suggestion
import { trpc } from "@lib/trpc";
```

File: apps/web/components/DestinationCalendarSelector.tsx [hunk 28]
```diff
@@ -67,6 +67,18 @@ const DestinationCalendarSelector = ({
         name={"primarySelectedCalendar"}
         placeholder={!hidePlaceholder ? `${t("select_destination_calendar")}:` : undefined}
         options={options}
+        styles={{
+          option: (defaultStyles: any, state: any) => ({
```
Comment (zomars) [inline apps/web/components/DestinationCalendarSelector.tsx:71]:
```suggestion
          option: (defaultStyles, state) => ({
```

File: apps/web/components/integrations/CalendarListContainer.tsx [hunk 29]
```diff
@@ -1,19 +1,20 @@
...
+import { useLocale } from "@calcom/lib/hooks/useLocale";
 import showToast from "@calcom/lib/notification";
 import { Alert } from "@calcom/ui/Alert";
 import Button from "@calcom/ui/Button";
 import Switch from "@calcom/ui/Switch";
 
 import { QueryCell } from "@lib/QueryCell";
-import { useLocale } from "@lib/hooks/useLocale";
-import { trpc } from "@lib/trpc";
+import { trpc, inferQueryOutput } from "@lib/trpc";
```
Comment (zomars) [inline apps/web/components/integrations/CalendarListContainer.tsx:11]:
```suggestion
import { trpc } from "@lib/trpc";
```

File: apps/web/pages/apps/installed.tsx [hunk 30]
```diff
@@ -96,122 +95,93 @@ function ConnectOrDisconnectIntegrationButton(props: {
...
+          <>
+            {data.items.length > 0 && (
+              <div className={className}>
+                <ShellSubHeading
+                  title={
+                    <SubHeadingTitleWithConnections title={t(variant)} numConnections={data.items.length} />
+                  }
+                />
+                <List>
+                  {data.items.map((item: AppOutput) => (
```
Comment (zomars) [inline apps/web/pages/apps/installed.tsx:124]:
```suggestion
                  {data.items.map((item) => (
```

File: apps/web/components/AdditionalCalendarSelector.tsx (hunk 1 above)
Comment (leog) [reply apps/web/components/AdditionalCalendarSelector.tsx:33]: Wouldn't work, the app is already installed, so the button is disabled to be able to click it to add/connect a new Apple Calendar:
Cal_com___Cal_com

Comment (zomars) [reply apps/web/components/AdditionalCalendarSelector.tsx:33]: We need to fix the dialog to merge then

File: packages/prisma/migrations/20220525182228_cal_video_preinstalled/migration.sql [hunk 31]
```diff
@@ -0,0 +1 @@
+insert into "App" ("slug", "dirName", "updatedAt", "categories") values ('daily-video', 'dailyvideo', NOW(), '{video}')
```
Comment (zomars) [inline packages/prisma/migrations/20220525182228_cal_video_preinstalled/migration.sql:1]: This migration is breaking builds. Removed `autoupdate`. Let's abstain to push until this is resolved.

**Review by zomars — DISMISSED** [review]

**Comment by leog** [comment]
Follow-up Loom video for the added changes in scope: https://www.loom.com/share/8531837b3eb44506bcd42b9eccdc357d

File: apps/web/components/DestinationCalendarSelector.tsx [hunk 32]
```diff
@@ -52,23 +68,29 @@ const DestinationCalendarSelector = ({
     })) ?? [];
   return (
     <div className="relative" title={`${t("select_destination_calendar")}: ${selectedOption?.label || ""}`}>
-      {/* There's no easy way to customize the displayed value for a Select, so we fake it. */}
```
Comment (leog) [inline apps/web/components/DestinationCalendarSelector.tsx:55]: Simplifying this part which was kind of hacky with a built-in way to handle it

File: apps/web/components/integrations/CalendarListContainer.tsx [hunk 33]
```diff
@@ -207,39 +174,54 @@ export function CalendarListContainer(props: { heading?: false }) {
...
-        <ShellSubHeading
-          className="mt-6"
-          title={<SubHeadingTitleWithConnections title={t("connect_an_additional_calendar")} />}
-        />
-      )}
-      <CalendarList onChanged={onChanged} />
-    </>
+    <QueryCell
+      query={query}
+      customLoader={<SkeletonLoader className="mt-10" />}
```
Comment (leog) [inline apps/web/components/integrations/CalendarListContainer.tsx:185]: Adding missing skeleton loader for better UX

File: apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx [hunk 34]
```diff
@@ -22,65 +24,63 @@ function ApiKeyListContainer() {
...
-                      apiKey={item}
-                      onEditApiKey={() => {
-                        setApiKeyToEdit(item);
-                        setEditModalOpen(true);
-                      }}
-                    />
-                  ))}
-                </List>
-              )}
+    <QueryCell
```
Comment (leog) [inline apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:27]: Also went for using standard Shells to handle this type of sections for consistency

File: apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx [hunk 35]
```diff
@@ -22,65 +24,63 @@ function ApiKeyListContainer() {
...
+      query={query}
+      customLoader={<SkeletonLoader />}
+      success={({ data }) => (
+        <div className="border-b border-gray-200 py-8 pl-2 pr-1">
+          <ShellSubHeading
```
Comment (leog) [inline apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:32]: Also applied the standard Shells we already have to handle this type of sections for consistency

File: apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx [hunk 36]
```diff
@@ -22,65 +24,63 @@ function ApiKeyListContainer() {
...
-          <h2 className="font-cal text-lg font-medium leading-6 text-gray-900">{t("api_keys")}</h2>
-          <p className="mt-1 mb-5 text-sm text-gray-500">{t("api_keys_subtitle")}</p>
-        </div>
-        <div className="mb-9 sm:self-center">
-          <Button StartIcon={PlusIcon} color="secondary" onClick={() => setNewApiKeyModal(true)}>
-            {t("generate_new_api_key")}
-          </Button>
-        </div>
-      </div>
-      <LicenseRequired>
```
Comment (leog) [inline apps/web/ee/components/apiKeys/ApiKeyListContainer.tsx:37]: Bad merge, fixing this!

File: packages/prisma/migrations/20220525182228_cal_video_preinstalled/migration.sql [hunk 37]
```diff
@@ -0,0 +1,3 @@
+insert into "App" ("slug", "dirName", "updatedAt", "categories") 
+values ('daily-video', 'dailyvideo', CURRENT_TIMESTAMP, '{video}') 
+on conflict do nothing
```
Comment (leog) [inline packages/prisma/migrations/20220525182228_cal_video_preinstalled/migration.sql:3]: This covers having Cal Video installed by default. Follow-up changes will tackle removing globally installed apps to also be able to uninstall them, no special treatment.

File: packages/ui/skeleton/index.tsx [hunk 38]
```diff
@@ -27,9 +27,17 @@ const SkeletonText: React.FC<SkeletonBaseProps> = ({ width, height }) => {
...
 };
 
+const SkeletonButton: React.FC<SkeletonBaseProps> = ({ width, height }) => {
+  return (
+    <SkeletonContainer>
+      <div className={`w-${width} h-${height} bg-gray-200 ${classNames}`} />
+    </SkeletonContainer>
+  );
+};
+
```
Comment (leog) [inline packages/ui/skeleton/index.tsx:37]: Needed a pulsing button for the loading Install button in App pages

File: packages/app-store/applecalendar/pages/setup/index.tsx [hunk 39]
```diff
@@ -0,0 +1,100 @@
...
+      password: "",
+    },
+  });
+
+  const [errorMessage, setErrorMessage] = useState("");
+
+  return (
+    <div className="flex h-screen bg-gray-200">
+      <div className="m-auto rounded bg-white p-5 md:w-[560px] md:p-10">
+        <div className="flex flex-col space-y-5 md:flex-row md:space-y-0 md:space-x-5">
```
Comment (leog) [inline packages/app-store/applecalendar/pages/setup/index.tsx:26]: New way to handle setup for apps, thanks for pointing to it @zomars. Made it responsive.

**Comment by hariombalhara** [comment]
Bug: Full page reloads when switching to installed apps for the first time
https://www.loom.com/share/08a0e1543184491daa169f1a96819fb4

File: apps/web/components/webhook/WebhookListItem.tsx (hunk 7 above)
Comment (hariombalhara) [inline apps/web/components/webhook/WebhookListItem.tsx:45]:
```suggestion
                    props.webhook.active ? "text-gray-200 bg-gray-200" : "bg-gray-50 text-gray-200"
```

File: apps/web/components/apps/SkeletonLoader.tsx [hunk 40]
```diff
@@ -4,10 +4,10 @@ import { SkeletonText } from "@calcom/ui";
 
 import { ShellSubHeading } from "@components/Shell";
 
-function SkeletonLoader() {
+function SkeletonLoader({ className }: { className?: string }) {
   return (
     <>
-      <ShellSubHeading title={<div className="h-6 w-32 rounded-sm bg-gray-100"></div>} />
+      <ShellSubHeading title={<div className="h-6 w-32 rounded-sm bg-gray-100"></div>} {...{ className }} />
```
Comment (hariombalhara) [inline apps/web/components/apps/SkeletonLoader.tsx:10]: Interesting way to add a property

**Review by hariombalhara — CHANGES_REQUESTED** [review]
Massive PR !! Awesome work @leog !!

Left a few comments, bugs are all together in this comment.
https://github.com/calcom/cal.com/pull/2751#issuecomment-1140969121

Feel free to address which are important.

File: apps/web/components/webhook/WebhookListContainer.tsx [hunk 41]
```diff
@@ -21,47 +20,35 @@ export type WebhookListContainerType = {
...
-      })
-    : trpc.useQuery(["viewer.webhook.list"], {
-        suspense: true,
-      });
-
+    ? trpc.useQuery(["viewer.webhook.list", { eventTypeId: props.eventTypeId }])
```
[the remark this answers is not in the document]
Comment (zomars) [reply apps/web/components/webhook/WebhookListContainer.tsx:24]: Agreed with @hariombalhara

**Comment by leog** [comment]
> **Bug-1**: Full page reloads when switching to installed apps for the first time

Fixed.

> **Bug-2**: Google Meet doesn't show up in Installed Apps when in fact I think it is installed at a global level and can't be uninstalled.

Fixed, marked Google Video (Meet) as global for now.

> **Bug-3**(On Production as well):
> The actual count of Other(6 and and Calendar apps is less than that of shown on App Store main page.

Doesn't happen for me, locally nor in production 🤔

> **Bug-4**: Disabling webhook doesn't gray out the entire tile

Fixed.

> **Suggestion-1**: How about adding a loader here. When the page is loading it feels like there is nothing here to load and it is broken.

Actually, I put a skeleton button there, pulsing to let the user know it will come up eventually. Maybe it's not enough?

https://user-images.githubusercontent.com/467258/171023589-0487c51d-0183-421c-84d4-08106baacf31.mov

File: apps/web/components/apps/SkeletonLoader.tsx (hunk 40 above)
Comment (leog) [reply apps/web/components/apps/SkeletonLoader.tsx:10]: haha I was kind of lazy it seems, and instead of duplicating the name of the property and its value, it came to me to do this

Comment (zomars) [reply apps/web/components/apps/SkeletonLoader.tsx:10]: It's ok to do this, but let's not fall into common pitfalls when doing this. I've seen how ugly this rabbit hole can go when abused.

File: apps/web/components/webhook/WebhookListContainer.tsx (hunk 41 above)
[the remark this answers is not in the document]
Comment (leog) [reply apps/web/components/webhook/WebhookListContainer.tsx]: Happy to help even if it's not code introduced by me hehe. Simplified.

**Review by hariombalhara — APPROVED** [review]

**Comment by Jaibles** [comment]
> Suggestion-2: Teams seem more important to me to come just next to Profile. Considering what you would use more frequently should come to the left side, it should not be after Developer or Security section. cc @Jaibles thoughts ?

Regarding this... happy for Teams to be the first thing after Profile.
